ipns/pb/Makefile

```diff
@@ -0,0 +1,31 @@
+# This Makefile is designed to work around namespace conflicts that occur when
+# multiple repositories depend on .proto files with generic filenames. Due to
+# the way protobuf registers files (e.g., foo.proto or pb/foo.proto), naming
+# collisions can occur when the same filename is used across different
+# packages.
+#
+# The only way to generate a *.pb.go file that includes the full package path
+# (e.g., github.com/org/repo/pb/foo.proto) is to place the .proto file in a
+# directory structure that mirrors its package path.
+#
+# References:
+# * https://protobuf.dev/reference/go/faq#namespace-conflict
+# * https://github.com/golang/protobuf/issues/1122#issuecomment-2045945265
+
+PKG_PATH = github.com/ipfs/boxo/ipns/pb
+PROTONAME = record
+PROTOFILE = ${PROTONAME}.proto
+
+all: clean build
+
+build:
+	mkdir -p ${PKG_PATH}
+	cp ${PROTOFILE} ${PKG_PATH} 
+	protoc --go_out=. ${PKG_PATH}/${PROTOFILE}
+	mv ${PKG_PATH}/${PROTONAME}.pb.go .
+	rm -rf $(firstword $(subst /, ,${PKG_PATH}))
+
+clean:
+	rm -f *.pb.go
+
+.PHONY: all build clean
```

Comment:
I think it would be preferable to do that as a go script instead of as a makefile. That would ensure compatibility across all platforms that have a go compiler. The go script would then be called using `go generate`, with this in `ipns/record.go`:
```
//go:generate go run build_proto.go

```
Another option, instead of a go script, the commands could also be called as a series of `//go:generate` directives.

@lidel Are there other examples of similar things in the IPFS ecosystem that we want emulate for consistency?